CMakeLists.txt

```diff
@@ -7,19 +7,86 @@ if (NOT SKBUILD)
                       "Please run\n$ pip install .")
 endif()
 
-# {{{ Constrain FindPython to find the Python version used by scikit-build
+# {{{ Options
+
+option(PYOPENCL_TRACE "Enable OpenCL tracing" OFF)
+option(PYOPENCL_ENABLE_GL "Enable OpenGL interoperability" OFF)
+option(PYOPENCL_USE_SHIPPED_EXT "Use shipped CL extension header" OFF)
+set(CL_INC_DIR CACHE STRING "OpenCL include directory")
+set(CL_LIB_DIR CACHE STRING "OpenCL library directory")
+set(CL_LIBNAME CACHE STRING "OpenCL library name")
+
+if(NOT CL_INC_DIR)
```

Comment:
I'm not sure I follow all this logic here, since it seems to overwrite the variables a few times. What's the priority order? I would expect something like
1. Whatever is passed to `cmake -DCL_LIB_DIR=/somedir` (or equivalently through pip `--config-settings`)
2. If not given, look at some standard environment variable like `OpenCL_ROOT` / `OCL_ROOT`
3. If not given, look at the environment variable `CL_LIB_DIR`.
5. If still not given, try to look at conda.
6. Let `FindOpenCL` try to find it.
7. Error!!!

It seems like this first tries to set it from `conda` (line 24) and then overwrites it with the environment variable (line 32).

Reply:
I think what you describe roughly matches what the implementation is (minus 2.). I'll clarify the code.

Reply:
[4ebf7e9](https://github.com/inducer/pyopencl/pull/755/commits/4ebf7e9468d2a13c829e82fe2ce30717acd0bcc3) clarifies this hopefully. The order of precedence currently is:

1. Whatever is passed as a CMake option, e.g. via `pip --config-settings`
2. If not given, look at the environment variables `CL_INC_DIR`, `CL_LIB_DIR`, `CL_LIBNAME`
3. If not, try looking at `CONDA_PREFIX`
4. If not, let `FindOpenCL` find it (this can work if OpenCL is installed system-wide, like on MacOS).
5. If not, error.

Reply:
This looks good to me now!

I only mentioned the `OCL_ROOT` variable because the version of `FindOpenCL` on my system (cmake `3.29.3`) has a `ENV OCL_ROOT` in the `find_library` hints, so I assumed that may be a more general variable.